src/models/model.py

```diff
@@ -239,6 +260,9 @@ def add_model_specific_args(parent_parser):
     print("model instance created")
 
     # train
-    trainer = pl.Trainer.from_argparse_args(args)
+    trainer = pl.Trainer(callbacks=[EarlyStopping(monitor='val_loss')]).from_argparse_args(args)
```

Comment:
without early stopping i have to manually kill jobs which means we'll never reach the predict/test step where we save predictions so added  lightning early stopping based on val loss. can also restrict the max epochs